base/int.jl

```diff
@@ -332,6 +332,21 @@ for to in BitInteger_types, from in (BitInteger_types..., Bool)
     end
 end
 
+"""
+    rem(x::Integer, T<:Integer)
+    mod(x::Integer, T<:Integer)
+    %(x::Integer, T<:Integer)
+
+Find `y::T` such that `x` ≡ `y` (mod n), where n is the number of integers representable
+in `T`, and `y` is an integer in `[typemin(T),typemax(T)]`.
```

Comment:
With this definition would it make sense to define (extend) this operation for `T == BigInt` ? (but somehow, till now I understood  `x % T` as a cheap bit-level cast operation, which would not work for `BigInt`...)